From 204181cfc115b03a42dba6cc6d545caf796ccf08 Mon Sep 17 00:00:00 2001
From: Patrick Rohr <prohr@google.com>
Date: Fri, 27 Jan 2023 14:40:43 -0800
Subject: [PATCH] cronet: fix proguard config to account for jarjar

All classes inside the cronet_aml_java.jar get jarjar prefixed with
android.net.http.internal.

Test: dexdump classes.dex | grep CollectionUtil
Change-Id: Icf8b426f3284472025c6530a2574ddae77717ccf
---
 .../android/cronet_impl_native_proguard.cfg   | 22 +++++++++----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/components/cronet/android/cronet_impl_native_proguard.cfg b/components/cronet/android/cronet_impl_native_proguard.cfg
index 49187704db4e9..ecea359a6370e 100644
--- a/components/cronet/android/cronet_impl_native_proguard.cfg
+++ b/components/cronet/android/cronet_impl_native_proguard.cfg
@@ -1,27 +1,27 @@
 # Proguard config for apps that depend on cronet_impl_native_java.jar.
 
 # Suppress unnecessary warnings.
--dontnote org.chromium.net.ProxyChangeListener$ProxyReceiver
--dontnote org.chromium.net.AndroidKeyStore
+-dontnote android.net.http.internal.org.chromium.net.ProxyChangeListener$ProxyReceiver
+-dontnote android.net.http.internal.org.chromium.net.AndroidKeyStore
 # Needs 'void setTextAppearance(int)' (API level 23).
--dontwarn org.chromium.base.ApiCompatibilityUtils
+-dontwarn android.net.http.internal.org.chromium.base.ApiCompatibilityUtils
 # Needs 'boolean onSearchRequested(android.view.SearchEvent)' (API level 23).
--dontwarn org.chromium.base.WindowCallbackWrapper
+-dontwarn android.net.http.internal.org.chromium.base.WindowCallbackWrapper
 
 # Generated for chrome apk and not included into cronet.
--dontwarn org.chromium.base.multidex.ChromiumMultiDexInstaller
--dontwarn org.chromium.base.library_loader.LibraryLoader
--dontwarn org.chromium.base.SysUtils
--dontwarn org.chromium.build.NativeLibraries
+-dontwarn android.net.http.internal.org.chromium.base.multidex.ChromiumMultiDexInstaller
+-dontwarn android.net.http.internal.org.chromium.base.library_loader.LibraryLoader
+-dontwarn android.net.http.internal.org.chromium.base.SysUtils
+-dontwarn android.net.http.internal.org.chromium.build.NativeLibraries
 
 # Objects of this type are passed around by native code, but the class
 # is never used directly by native code. Since the class is not loaded, it does
 # not need to be preserved as an entry point.
--dontnote org.chromium.net.UrlRequest$ResponseHeadersMap
+-dontnote android.net.http.internal.org.chromium.net.UrlRequest$ResponseHeadersMap
 # https://android.googlesource.com/platform/sdk/+/marshmallow-mr1-release/files/proguard-android.txt#54
--dontwarn android.support.**
+-dontwarn android.net.http.internal.android.support.**
 
 # This class should be explicitly kept to avoid failure if
 # class/merging/horizontal proguard optimization is enabled.
--keep class org.chromium.base.CollectionUtil
+-keep class android.net.http.internal.org.chromium.base.CollectionUtil
 
-- 
2.39.1.456.gfc5497dd1b-goog

